pwiz_tools/BiblioSpec/src/BlibBuilder.cpp

```diff
@@ -566,72 +580,77 @@ int BlibBuilder::readSequences(set<string>** seqSet, bool modified)
             break;
         }
 
-        string newSeq;
-        string unexpected;
-        vector<SeqMod> mods;
-        size_t aaPosition = 0;
-        // check that each character is a letter and convert it to uppercase
-        for (size_t i = 0; i < sequence.length(); ++i)
+        string newSeq = parseSequence(sequence, modified);
+        Verbosity::debug("Adding target sequence %s", newSeq.c_str());
+        (*seqSet)->insert(newSeq);
+        ++sequencesRead;
+    }
+
+    return sequencesRead;
+}
+
+string BlibBuilder::parseSequence(const string& sequence, bool modified) {
+    string newSeq;
+    string unexpected;
+    vector<SeqMod> mods;
+    size_t aaPosition = 0;
+    // check that each character is a letter and convert it to uppercase
+    for (size_t i = 0; i < sequence.length(); ++i)
+    {
+        if (isalpha(sequence[i]))
         {
-            if (isalpha(sequence[i]))
-            {
-                newSeq += toupper(sequence[i]);
-                ++aaPosition;
-            }
-            else if (modified && sequence[i] == '[')
+            newSeq += toupper(sequence[i]);
+            ++aaPosition;
+        }
+        else if (modified && sequence[i] == '[')
+        {
+            // get modification
+            size_t endIdx = sequence.find(']', i + 1);
+            if (endIdx != string::npos)
             {
-                // get modification
-                size_t endIdx = sequence.find(']', i + 1);
-                if (endIdx != string::npos)
+                ++i;
+                istringstream deltaMassExtractor(sequence.substr(i, endIdx - i));

```

Comment:
How many rows are you likely to be parsing? If it's anything significant, you should use lexical_cast<double> here (which is overloaded to use strtod, much more efficient than istringstream).